addons/project/tests/test_project_sharing.py

```diff
@@ -295,3 +297,39 @@ def test_orm_method_with_true_false_domain(self):
             [],
         )
         self.assertFalse(task_read_group[0]['__count'], 'No result should found with the read_group since the domain is falsy.')
+
+    def test_add_followers_from_share_edit_wizard(self):
+        """
+            This test ensures that when a project is shared in edit mode, the partners are correctly set as follower in the project and their respective tasks.
+        """
+        partner_a, partner_b, partner_d = self.env['res.partner'].create([
+            {'name': "Solanum", 'parent_id': self.env.company.partner_id.id},
+            {'name': "Zana", 'parent_id': self.env.company.partner_id.id},
+            {'name': "Thresh"},
+        ])
+        partners = partner_a | partner_b | partner_d | self.env.company.partner_id
```

Comment:
```suggestion
        company_partner = self.env.company.partner_id
        partners = partner_a, partner_b, partner_d = self.env['res.partner'].create([
            {'name': "Solanum", 'parent_id': company_partner.id},
            {'name': "Zana", 'parent_id': company_partner.id},
            {'name': "Thresh"},
        ])
        partners |= company_partner
```
You will use `company_partner` later too.